### PR TITLE
added custom wrapper for jax.lax.top_k in new file

### PR DESCRIPTION
**Pull request by rohan-mehta-1024**
## Description
Added a custom wrapper for jax.lax.top_k in a new file `specialized_fns.py` – probably a bad name, but did not know what else to use. Should likely be renamed in the future. 

## Fixes Issues
#32 

## Unit test coverage
Tests the function over each axis of a random tensor and compares against the corresponding output for jax.lax.top_k.

## Known breaking changes/behaviors
N/A

### CODE REVIEW

File: src/haliax/specialized_fns.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+def top_k(arr: NamedArray, axis: AxisSelector, k: int) -> NamedArray:
+    pos = arr._lookup_indices(axis)
+    if pos is None:
+        raise ValueError(f"Axis {axis} not found in {arr}")
+    new_array = jnp.moveaxis(arr.array, pos, -1)  # move axis to the last position
+    values, _ = jax.lax.top_k(new_array, k=k)
+    values = jnp.moveaxis(values, -1, pos)  # move axis back to its original position
+
+    axis = arr.resolve_axis(axis)
+    new_axis = Axis(f"top_{k}_of_{axis.name}", size=k)
```
Comment (dlwh) [inline src/haliax/specialized_fns.py:17]: imho i think it's better to just keep the same name and resize it, or even better:

```
def top_k(arr: NamedArray, axis: AxisSelector, k: int, new_axis: Optional[AxisSelector] = None) -> NamedArray:
  ...
  if new_axis is None: 
       new_axis = axis.resize(k)
  else:
      ... 
 
```

Otherwise it's a pain to refer to the top-k axis

**Review by dlwh — CHANGES_REQUESTED** [review]
thanks! I think it's better if we make the name something configurable, and make the default be something a bit more accessible.

File: src/haliax/specialized_fns.py (hunk 1 above)
Comment (rohan-mehta-1024) [reply src/haliax/specialized_fns.py]: Yup, this makes sense – should be fixed now!

File: src/haliax/__init__.py [hunk 2]
```diff
@@ -36,6 +36,7 @@
 from .hof import fold, map, scan, vmap
 from .ops import clip, isclose, pad_left, trace, tril, triu, where
 from .partitioning import auto_sharded, axis_mapping, fsdp, named_jit, shard_with_axis_mapping
+from .specialized_fns import top_k
```
Comment (dlwh) [inline src/haliax/__init__.py:39]: you should add top_k to __all__. sorry i missed that.

**Review by dlwh — APPROVED** [review]
1 tiny thing, then all set. Thanks!

File: src/haliax/__init__.py (hunk 2 above)
Comment (rohan-mehta-1024) [reply src/haliax/__init__.py:39]: Ah, didn't realize I missed that. Just added it.
